packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -774,6 +789,38 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		}
 	}
 
+	/**
+	 * Delete unused storage blobs, if any. After GC runs and notifies blob manager of local blob ids that are sweep
+	 * ready, blob manager deletes them from the redirect table. This function is then called to delete the storage
+	 * ids of these blobs if they are not used anymore. A blob is unused if there is no entry from the local id to
+	 * storage id in the redirect table, i.e., all local id entries have been deleted.
+	 * @param maybeUnusedBlobIds - List of storage ids for blobs whose local id entries were just deleted and are
+	 * now candidates for deletion.
+	 */
+	private deleteUnusedStorageBlobs(maybeUnusedBlobIds: string[]) {
+		if (maybeUnusedBlobIds.length === 0) {
+			return;
+		}
+
+		const storageBlobsToDelete: Set<string> = new Set(maybeUnusedBlobIds);
+		for (const [localId, storageId] of this.redirectTable.entries()) {
+			// For a storage id that is a candidate for deletion, ignore storageId -> storageId entries. But if the
+			// redirect table contains an entry from a local id to it, it is still in use and should not be deleted.
+			if (
+				storageId !== undefined &&
```

Comment:
When does this happen?  And wouldn't the next line filter out undefined anyway?

Reply:
Oh I found it - it's undefined when detached.  Is it too risky to assert that storageId isn't undefined here?